plasmapy/particles/_factory.py

```diff
@@ -9,15 +9,105 @@
 import astropy.units as u
 import contextlib
 
-from astropy.units.physical import electrical_charge, mass
-from numbers import Integral
-from typing import Union
+from astropy.constants import m_e
+from numbers import Integral, Real
+from typing import Any, Dict, Union
 
 from plasmapy.particles.exceptions import InvalidParticleError
 from plasmapy.particles.particle_class import CustomParticle, Particle
 from plasmapy.particles.particle_collections import ParticleList
 
 
+def _generate_error_message(args: tuple, kwargs: Dict[str, Any]) -> str:
+    """Compose an error message for invalid particles."""
+
+    errmsg = "Unable to create a particle from: "
+
+    if args:
+        errmsg += repr(args)
+        if kwargs:
+            errmsg += " and "
+
+    if kwargs:
+        errmsg += repr(kwargs)
+
+    errmsg += (
+        ". For information on creating particles, see: "
+        "https://docs.plasmapy.org/en/stable/glossary.html"
+    )
+
+    return errmsg
+
+
+def _make_custom_particle_with_real_charge_number(*args, **kwargs):
+    """
+    Create a |CustomParticle| for mean or composite ions.
+
+    This function is intended to produce |CustomParticle| instances
+    provided a string representing an element or isotope without charge
+    information (e.g., ``"He"`` or ``"He-4"`` but not ``"He-4 2+"``)
+    along with a charge number that is a real number but not an integer.
```

Comment:
I'm thinking it would be worthwhile to make this capability (along with the rest of the particle factory functions) part of our public API, but I want to be able to freely change the API while getting some practice using them.

Reply:
w i s d o m

No, seriously, it's just hit me that we've come a long way from the early days 😀

Reply:
Hm...I probably shouldn't mention that my true motivation was trying to find a way to justify being lazy! 😹